internal/version/release_version.go

```diff
@@ -40,15 +41,36 @@ type releaseVersionFetcher struct {
 	ctx context.Context
 }
 
-// LatestVersion encapsulates the logic to manage different cloud providers.
-func (s *releaseVersionFetcher) LatestVersion() (*ReleaseInformation, error) {
+// LatestWithCriteria retrieves the first release version that matches the criteria. We assume that ListReleases returns releases sorted by created_at value.
+func (s *releaseVersionFetcher) LatestWithCriteria(n int, matchCriteria Criteria, toolName string) (*github.RepositoryRelease, error) {
+	var pageSize = n
+
+	startTime := time.Now()
 	client := github.NewClient(nil)
-	release, _, err := client.Repositories.GetLatestRelease(s.ctx, "mongodb", "mongocli")
-	if err != nil {
-		return nil, err
+
+	opt := &github.ListOptions{PerPage: pageSize}
+
+	if matchCriteria == nil {
+		matchCriteria = func(tag, tool string) bool {
+			return true
+		}
+	}
+
+	for {
+		releases, resp, err := client.Repositories.ListReleases(s.ctx, owner, MongoCLI, opt)
+		if err != nil {
+			return nil, err
+		}
+		// Returns as soon as criteria is matched
+		for _, release := range releases {
+			if matchCriteria(release.GetTagName(), toolName) && !release.GetPrerelease() {
```

Comment:
I intentionally added this logic since I think it's more likely to see Releases than PreReleases. (I see no prereleases here https://api.github.com/repos/mongodb/mongocli/releases)

Reply:
Good point! Thanks! : )